src/ui/public/directives/file_upload.js

```diff
@@ -1,32 +1,80 @@
+import _ from 'lodash';
 import $ from 'jquery';
 import uiModules from 'ui/modules';
 var module = uiModules.get('kibana');
 
-module.directive('fileUpload', function ($parse) {
+var html = '<span class="dropzone" ng-transclude></span>';
+
+module.directive('fileUpload', function () {
   return {
-    restrict: 'A',
+    restrict: 'E',
+    transclude: true,
+    scope: {
+      onRead: '&',
+      onLocate: '&',
+      label: '='
+    },
+    template: html,
     link: function ($scope, $elem, attrs) {
-      var onUpload = $parse(attrs.fileUpload);
+      var $button = $elem.find('.upload');
```

Comment:
I thought about this more, and I thought, what if there was something like

```javascript
scope: {
  onRead: '&',
  onLocate: '&',
  label: '=',
  uploadSelector: '@'
}
```

and then

```javascript
var $button = $elem.find($scope.uploadSelector);
```

Thoughts? I would just prefer it be a more explicit part of the API of the directive.